packages/governance/src/governor/governor.cairo

```diff
@@ -0,0 +1,967 @@
+        fn state(self: @ComponentState<TContractState>, proposal_id: felt252) -> ProposalState;
+
+        /// Address through which the governor executes action.
+        /// Should be used to specify whether the module execute actions through another contract
+        /// such as a timelock.
+        fn executor(self: @ComponentState<TContractState>) -> ContractAddress;
+
+        /// Execution mechanism. Can be used to modify the way execution is
+        /// performed (for example adding a vault/timelock).
+        fn execute_operations(
+            ref self: ComponentState<TContractState>,
+            proposal_id: felt252,
+            calls: Span<Call>,
+            description_hash: felt252
+        );
+
+        /// Queuing mechanism. Can be used to modify the way queuing is
+        /// performed (for example adding a vault/timelock).
+        ///
+        /// Requirements:
+        ///
+        /// - Must return a timestamp that describes the expected ETA for execution. If the returned
+        /// value is 0, the core will consider queueing did not succeed, and the public `queue`
+        /// function will revert.
+        fn queue_operations(
+            ref self: ComponentState<TContractState>,
+            proposal_id: felt252,
+            calls: Span<Call>,
+            description_hash: felt252
+        ) -> u64;
+
+        /// See `interface::IGovernor::proposal_needs_queuing`.
+        fn proposal_needs_queuing(
+            self: @ComponentState<TContractState>, proposal_id: felt252
+        ) -> bool;
+
+        /// Cancel mechanism. Can be used to modify the way canceling is
+        /// performed (for example adding a vault/timelock).
+        fn cancel_operations(
+            ref self: ComponentState<TContractState>,
+            proposal_id: felt252,
+            description_hash: felt252
+        );
+    }
+
+    //
+    // External
+    //
+
+    #[embeddable_as(GovernorImpl)]
+    impl Governor<
+        TContractState,
+        +HasComponent<TContractState>,
+        +SRC5Component::HasComponent<TContractState>,
+        +GovernorVotesTrait<TContractState>,
+        impl GovernorQuorum: GovernorQuorumTrait<TContractState>,
+        impl GovernorCounting: GovernorCountingTrait<TContractState>,
+        impl GovernorExecution: GovernorExecutionTrait<TContractState>,
+        impl GovernorSettings: GovernorSettingsTrait<TContractState>,
+        impl Metadata: SNIP12Metadata,
+        impl Immutable: ImmutableConfig,
+        +Drop<TContractState>,
+    > of IGovernor<ComponentState<TContractState>> {
+        /// Name of the governor instance (used in building the SNIP-12 domain separator).
+        fn name(self: @ComponentState<TContractState>) -> felt252 {
+            Metadata::name()
+        }
+
+        /// Version of the governor instance (used in building SNIP-12 domain separator).
+        fn version(self: @ComponentState<TContractState>) -> felt252 {
+            Metadata::version()
+        }
```

Comment:
In that case, we have already a `name` function in the contract, and a standard Permit interface not containing name and version. In this case (governor), we don't have a name so we can use name and version for that, and we don't have a standard interface, so I don't think is worth separating the interface, since `cast_vote_by_sig` params require access to this info and is part of IGovernor.

Reply:
Also this https://github.com/OpenZeppelin/cairo-contracts/pull/1180#discussion_r1838246048

Reply:
Mmm right, there's the name clash with permit and this isn't standardized. Keeping it as is sounds good to me!